ext/gd/gd.c

```diff
@@ -1499,8 +1499,12 @@ static int _php_image_type (zend_string *data)
 
 	php_stream *image_stream = php_stream_memory_open(TEMP_STREAM_READONLY, data);
 
-  if (php_is_image_avif(image_stream)) {
-  	return PHP_GDIMG_TYPE_AVIF;
+  if (image_stream != NULL) {
+		if (php_is_image_avif(image_stream)) {
+  		return PHP_GDIMG_TYPE_AVIF;
+		}
+
+		php_stream_close(image_stream);
 	}
```

Comment:
Unless `php_is_image_avif()` closes the stream on success (what it probably shouldn't), we need to close the stream ourselves in this case. Maybe:
````
	if (image_stream != NULL) {
		bool is_avif = php_is_image_avif(image_stream);
		php_stream_close(image_stream);
		if (is_avif) {
			return PHP_GDIMG_TYPE_AVIF;
		}
	}